dash/testing/dash_page.py

```diff
@@ -42,9 +42,17 @@ def window_store(self):
 
     def _wait_for_callbacks(self):
         if self.window_store:
-            return self.redux_state_rqs and all(
-                (_.get("responseTime") for _ in self.redux_state_rqs)
+            # note that there is still a small chance of Negative True
+            # where we get two earlier requests in the queue, this returns
+            # True but there are still more requests to come
+            return all(
+                (
+                    _.get("responseTime")
+                    for _ in self.redux_state_rqs
+                    if _.get("controllerId")
```

Comment:
ha, I think I see what's going on here, thanks for bringing this up. Interestingly this was a question from the beginning when multi-outputs was implemented:

https://github.com/plotly/dash/blob/046c438928c17e0f7faf5343f5f4e6d98d0c9efd/dash-renderer/src/actions/index.js#L933-L934

Yes, there seem to be some implications... probably due to the fact that `dispatch` is not synchronous. Not clear to me why we need this dummy request in the first place, but I'll figure that out.

Reply:
For followup in #1008

Reply:
another issue in the same basket is #992